src/Utilities/Wrapper.js

```diff
@@ -9,10 +9,44 @@ const RenderWrapper = ({
   store,
   ...props
 }) => {
-  const { hasAccess } = usePermissionsWithContext([
-    'inventory:*:read',
-    'inventory:hosts:read',
-  ]);
+  const [allUserPermissions, setAllUserPermissions] = useState();
+  const chrome = useChrome();
+
+  //this is used to get all user permissions without HOOKS
+  const getUserRbacPermissions = () => {
+    try {
+      return chrome.getUserPermissions('inventory');
+    } catch (e) {
+      throw new Error(`Error getting user permissions: ${e.message}`);
+    }
+  };
+
+  //we get the data from the chrome.getUserPermissions and map it to object named user
+  //then we return a user object
+  function mapRbacPermissionsToUser(permissionArray) {
+    const user = {};
+
+    for (const obj of permissionArray) {
+      // Extract the permission name without colons and in camel case
+      const permissionName = obj.permission;
+
+      user[permissionName] = true;
+    }
+
+    return user;
+  }
+
+  //this useEffect trigger once on the page load, get user permissions from the promise
+  //then trigger mapping function and set's user permission data to useState
+  //that way we can get the data from the rbac early and work with the group ID's before we request API or call Hooks
```

Comment:
I don't see a real value in such extensive comments: basically, the code describes itself pretty well already.